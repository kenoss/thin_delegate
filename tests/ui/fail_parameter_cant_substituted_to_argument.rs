#[thin_delegate::register]
trait Hello<'a, T> {
    fn hello(&self) -> &'a T;
}

#[thin_delegate::register]
struct Hoge;

#[thin_delegate::derive_delegate]
impl Hello<String, String> for Hoge {}

fn main() {}
