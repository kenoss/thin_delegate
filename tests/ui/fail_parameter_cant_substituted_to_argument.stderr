error: parameter can't be substituted to argument:
           in definition of trait         = < 'a, T >
           in definition of derive target = < String, String >
  --> tests/ui/fail_parameter_cant_substituted_to_argument.rs:10:12
   |
10 | impl Hello<String, String> for Hoge {}
   |            ^^^^^^
